src/schemaGenerator.js

```diff
@@ -48,7 +49,7 @@ function _generateSchema(
 
   addResolveFunctionsToSchema(schema, resolveFunctions);
 
-  assertResolveFunctionsPresent(schema);
+  assertResolveFunctionsPresent(schema, allowDefaultResolve);
```

Comment:
Rather than passing in `allowDefaultResolve` to `assertResolveFunctionsPresent`, how about we just have an on/off switch? I think that will be easier for folks to understand because either they get all of the checks or they get none. There will be other ways to check arguments, and I can already foresee someone wanting to turn off checks for resolve functions present in resolvers but not in the schema. Having too many fine-grained controls may make it harder for people to understand what's really going to happen when they toggle a combination of options. 

If we do that, we can just call the argument `checkResolvers`, and do something like this:
```
if (checkResolvers) {
  assertResolveFunctionsPresent(schema);
}
```

While we're at it, we should also pass that in to addResolveFunctionsToSchema, and turn off the checks there that make sure the types / fields defined in the resolvers actually exist.

What do you think? Happy to discuss this further!